### PR TITLE
[Minecraft] Use DS annotations, fix static code analysis findings

### PR DESCRIPTION
**Pull request by wborn**
* Use DS annotations
* Fix static code analysis findings

### CODE REVIEW

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java [hunk 1]
```diff
@@ -78,6 +84,12 @@ private void discoverServers() {
         subscription.add(signsSubscription);
     }
 
+    @Modified
+    @Override
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: @wborn why?

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java [hunk 2]
```diff
@@ -150,11 +162,8 @@ private void submitPlayerDiscoveryResults(ThingUID bridgeUID, String name) {
 
         Map<String, Object> properties = new HashMap<>();
         properties.put(MinecraftBindingConstants.PARAMETER_PLAYER_NAME, name);
-        thingDiscovered(DiscoveryResultBuilder.create(uid)
-                .withProperties(properties)
-                .withBridge(bridgeUID)
-                .withLabel(name)
-                .build());
+        thingDiscovered(DiscoveryResultBuilder.create(uid).withProperties(properties).withBridge(bridgeUID)
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: pitty ;-)

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/server/ServerConnection.java [hunk 3]
```diff
@@ -31,11 +30,11 @@
 /**
  * Holds information about connection to Minecraft server.
  *
- * @author Mattias Markehed
+ * @author Mattias Markehed - Initial contribution
  */
 public class ServerConnection {
 
-    private static final Logger logger = LoggerFactory.getLogger(ServerConnection.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServerConnection.class);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/server/ServerConnection.java]: Does it need to be static?

Comment (martinvw) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/server/ServerConnection.java]: If possible not is what I mean :-)

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java (hunk 1 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: It was also part of the [original XML](https://github.com/openhab/openhab2-addons/blob/242c956d4e32a3cd240c558109f65da243245e37/addons/binding/org.openhab.binding.minecraft/OSGI-INF/MinecraftDiscovery.xml#L12). 
So I added the annotation for generating equivalent XML.

Perhaps @ibaton knows if it was intentionally added to the orignal XML?

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java (hunk 2 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: That part of the contribution was by the formatter and not me. :innocent:

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/server/ServerConnection.java (hunk 3 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/server/ServerConnection.java]: I've managed to move the logger to the object where it is used so it is no longer static.

File: addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java (hunk 1 above)
Comment (ibaton) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: I don't think it was intended, I have no memory of adding it intentionally.

Comment (wborn) [reply addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/internal/discovery/MinecraftDiscoveryService.java]: Thanks for the info! I've removed the `modified(..)` method and annotation in a new commit.

**Comment by ibaton** [comment]
:+1: Thanks for cleaning it up.

**Pull request merged**